force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -271,11 +338,18 @@ public with sharing class FlowControlLoopsCollections {
      */
     public static Integer findWordInList(String wordToFind, List<String> words) {
         // The variable to store the index of the word
+        Integer indexOfWord = words.indexOf(wordToFind);
```

Comment:
Using the indexOf() method really simplifies the code required here. It makes the for loop on line 344 redundant as you are already capturing the index of that word.